opentracing/src/main/java/opentracing/TraceContextSource.java

```diff
@@ -1,48 +0,0 @@
-/**
- * Copyright 2016 The OpenTracing Authors
- *
- * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
- * in compliance with the License. You may obtain a copy of the License at
- *
- * http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software distributed under the License
- * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
- * or implied. See the License for the specific language governing permissions and limitations under
- * the License.
- */
-package opentracing;
-
-import java.util.Map;
-
-/**
- * Long-lived interface that knows how to create a root {@linkplain TraceContext} and encode/decode
- * any other.
- */
-public interface TraceContextSource {
-
-  /**
-   * Encodes or Decodes a {@linkplain TraceContext trace context} in binary or text formats.
-   */
-  TraceContextCodec codec();
-
-  /**
-   * Create a TraceContext which has no parent (and thus begins its own trace).
-   *
-   * <p>A TraceContextSource must always return the same type in successive calls to
-   * NewRootTraceContext().
-   */
-  TraceContext newRoot();
-
-  /**
-   * Creates a child context for {@code parent}, and returns both that child's own
-   * TraceContext as well as any Tags that should be added to the child's Span.
-   */
-  ChildTraceContext newChild(TraceContext parent);
-
-  interface ChildTraceContext {
```

Comment:
this was c&p from go. basically there was an idea that you want to seed a child with a set of tags. Since tags aren't in a tracecontext, this is an issue.

This particular issue is implementation-specific.. ex in [Census Tracing](https://github.com/grpc/grpc/pull/4750), tags are in the same set, regardless of whether they are propagated or not. There certainly may be a filtering constraint when a tracing api is layered on that (ex is a child a fork of the parents tags, just changing id?) Maybe @bogdandrutu has ideas on how they will handle this.

In zipkin today, I don't know if any tracers propagate tags implicitly from parent to child, as a function of creating a child. Only exception I can think of is the endpoint of an annotation, but that's not actually scoped to the parent, more often the tracer itself.

Anyway, maybe someone else can elaborate on this feature..